test/e2e/sagemaker/tests/test_hpo.py

```diff
@@ -0,0 +1,151 @@
+        return ["Stopped", "Stopping"]
+
+    def _get_sagemaker_hpo_job_arn(self, sagemaker_client, hpo_job_name: str):
+        try:
+            hpo_desc = sagemaker_client.describe_hyper_parameter_tuning_job(
+                HyperParameterTuningJobName=hpo_job_name
+            )
+            return hpo_desc["HyperParameterTuningJobArn"]
+        except BaseException:
+            logging.error(
+                f"SageMaker could not find an hpo job with the name {hpo_job_name}"
+            )
+            return None
+
+    def _get_sagemaker_hpo_job_status(
+        self, sagemaker_client, hpo_job_name: str
+    ):
+        try:
+            hpo_job = sagemaker_client.describe_hyper_parameter_tuning_job(
+                HyperParameterTuningJobName=hpo_job_name
+            )
+            return hpo_job["HyperParameterTuningJobStatus"]
+        except BaseException:
+            logging.error(
+                f"SageMaker could not find an hpo job with the name {hpo_job_name}"
+            )
+            return None
+
+    def test_create_hpo(self, xgboost_hpojob):
+        (reference, resource) = xgboost_hpojob
+        assert k8s.get_resource_exists(reference)
+
+    def test_hpo_has_correct_arn(self, sagemaker_client, xgboost_hpojob):
+        (reference, _) = xgboost_hpojob
+        resource = k8s.get_resource(reference)
+        hpo_job_name = resource["spec"].get("hyperParameterTuningJobName", None)
+
+        assert hpo_job_name is not None
+
+        assert k8s.get_resource_arn(resource) == self._get_sagemaker_hpo_job_arn(
+            sagemaker_client, hpo_job_name
+        )
+
+    def test_hpo_job_has_created_status(
+        self, sagemaker_client, xgboost_hpojob
+    ):
+        (reference, _) = xgboost_hpojob
+        resource = k8s.get_resource(reference)
+        hpo_job_name = resource["spec"].get("hyperParameterTuningJobName", None)
+
+        assert hpo_job_name is not None
+
+        current_hpo_job_status = self._get_sagemaker_hpo_job_status(
+            sagemaker_client, hpo_job_name
+        )
+        expected_hpo_job_status_list = (
+            self._get_created_hpo_job_status_list()
+        )
+        assert current_hpo_job_status in expected_hpo_job_status_list
+
+    def test_hpo_job_has_stopped_status(
+        self, sagemaker_client, xgboost_hpojob
+    ):
+        (reference, _) = xgboost_hpojob
+        resource = k8s.get_resource(reference)
+        hpo_job_name = resource["spec"].get("hyperParameterTuningJobName", None)
+
+        assert hpo_job_name is not None
+
+        # Delete the k8s resource.
+        _, deleted = k8s.delete_custom_resource(reference)
+        assert deleted is True
+
+        current_hpo_job_status = self._get_sagemaker_hpo_job_status(
+            sagemaker_client, hpo_job_name
+        )
+        expected_hpo_job_status_list = (
+            self._get_stopped_hpo_job_status_list()
+        )
+        assert current_hpo_job_status in expected_hpo_job_status_list
```

Comment:
Seems a bit wasteful to me to have a bunch of test methods and call the SageMaker DescribeHyperParameterTuningJob API call multiple times just to test different attributes are set on the CR's Status struct.

You could also pull the `_get_sagemaker_hpo_job()` method out into a module-level function, decorate it with `@pytest.fixture` like so, and eliminate the need for the caller to pass the sagemaker_client param.

Is there a reason we couldn't combine these test methods into a single test? Something like this?

```python
@pytest.fixture(scope="module")
def get_sagemaker_hpo_job(self, sagemaker_client, hpo_job_name: str):
    try:
        hpo_desc = sagemaker_client.describe_hyper_parameter_tuning_job(
            HyperParameterTuningJobName=hpo_job_name
        )
        return hpo_desc
    except BaseException:
        logging.error(
            f"SageMaker could not find an hpo job with the name {hpo_job_name}"
        )
        return None

@service_marker
@pytest.mark.canary
class TestHPO:
    def test_smoke(self, xgboost_hpojob):
        (reference, resource) = xgboost_hpojob
        assert k8s.get_resource_exists(reference)

        hpo_job_name = resource["spec"].get("hyperParameterTuningJobName", None)
        assert hpo_job_name is not None

        hpo_job = self._get_sagemaker_hpo_job(sagemaker_client, hpo_job_name)

        assert k8s.get_resource_arn(resource) == hpo_job["HyperParameterTuningJobArn"]
        assert hpo_job["HyperParameterTuningJobStatus"] in HPO_JOB_STATUS_CREATED

        _, deleted = k8s.delete_custom_resource(reference)
        assert deleted is True

        assert hpo_job["HyperParameterTuningJobStatus"] in HPO_JOB_STATUS_STOPPED
```

Reply:
we cannot call `get_sagemaker_hpo_job ()` if we make that a fixture.

if we don't make it a fixture we also cannot use sagemaker_client as parameter.

so making both of these functions just normal functions instead of fixtures